docs/docs/adding-search-with-algolia.md

```diff
@@ -2,593 +2,612 @@
+The file exports a list of queries. Each query defines a single index. You can build [multiple indices](https://www.algolia.com/doc/guides/sending-and-managing-data/prepare-your-data/in-depth/choosing-between-one-or-more-indices/) with Algolia but this tutorial will only use a single one.
 
-Transformers allow you to modify the data returned by the queries to bring it into a format ready for searching. All you're doing here is 'flattening' posts and pages to 'unnest' the frontmatter fields (such as `author`, `date`, `tags`) but transformers could do much more for you if required. This makes the whole process of indexing your data really flexible and powerful. You could for instance use them to filter the results of your queries, format fields, add or merge them, etc.
+Each index requires a GraphQL query that retrieves the pages and data to be indexed. A `transformer` transforms the GraphQL data to an Algolia record.
 
-If you've come this far, then the "backend" is done. You should now be able to run `gatsby build` and see your indices in Algolia's web interface be flooded with your data.
+Note that each record must have an ID in the key `objectID`. The Algolia documentation provides more information on [how to structure data into records](https://www.algolia.com/doc/guides/sending-and-managing-data/prepare-your-data/#attributes---what-to-put-in-your-record).
 
-## Adding a search interface to your site
+Here, only the frontmatter field `title` and the field `excerpt` are indexed. We will display both fields in the search results. To index more fields, just add them to `pageQuery`.
 
-Next, build a user-facing search interface for your site. It needs a way for the user to enter a search string, send that string to Algolia, receive matching results (_hits_ in Algolia speak) from your indices and finally display those to the user.
+Each query has optional [settings](https://www.algolia.com/doc/api-reference/settings-api-parameters/). The code above tells Algolia we will want to generate "snippets" of context around our hits in the `excerpt` attribute.
 
-You're going to assemble everything you need into a React `Search` component that you call from anywhere on your site where you want the user to be able to search. Even though design varies strongly from site to site, you'll note the styles implemented with [`styled-components`](https://styled-components.com) in this guide since working out the CSS transitions to have the search field slide out as the user clicks on it and the results pane to appear once Algolia returns matches took some time.
+### Test your indexing
 
-The `Search` components is made up of the following files:
+This should complete the indexing setup. Now run `gatsby build`. If all goes well, the output should include the following:
```

Comment:
When I try to run a build at this point with a new Algolia account, I get this error:

```
AlgoliaSearchError: Method not allowed with this API key
```
Is there some kind of limit with free accounts, or something else I need to set up?

Reply:
What permissions does your search-only API key have? I didn't have the ability to edit it, so I made a new one and gave it search and browse permissions...but it still gave me the same error. I added a few more that seemed related (addObject, editSettings, usage) and no difference. I'm using the Application ID and search-only API key (or this new custom key) from my new Algolia account.

Reply:
I think I figured it out: I think you added the "Search-Only API Key
" as ALGOLIA_ADMIN_KEY. The search-only key only has read permissions but the admin key needs write permission since it's used for indexing. 

I get the same message as you (`AlgoliaSearchError: Method not allowed with this API key`) if I swap them.

If this is the issue I'll add it to the troubleshooting section!

Reply:
I think this should go better into the section „### Configuring the Algolia plugin“ where you write about setting up an API key. There you can write which permission a key needs.

 If I am a new user and have to setup permissions it takes always to much time to look into docs and/or google for infos which are the minimal needed settings.

Reply:
There shouldn't be any need to configure anything; you only need to create an account. The keys are all generated when you create the account. 

But we need to explain what that message means in case you put the keys in the wrong environment variables. (If Marcy verifies this was the issue)

Reply:
That seems to have been the issue.... I've been having problems with dotenv on my machine so I put the api keys directly in `gatsby-config`, and `apiKey` wasn't specific enough I guess! I've left comments on the other errors being thrown now that I seem to have gotten past this step. thanks!